src/python/pants/backend/python/lint/ruff/rules.py

```diff
@@ -64,7 +64,8 @@ async def run_ruff(
     )
 
     conf_args = [f"--config={ruff.config}"] if ruff.config else []
-    initial_args = ("--fix",) if request.is_fix else ()
+    # We must set `--force-exclude` so ruff applies excludes from config as designed by ruff
```

Comment:
Not really sure what "as designed by ruff" means? The important point to make here is that we enumerate individual files on the ruff command line, and ruff only applies configured excludes to those if we set that flag. (Otherwise the configured excludes only apply when you give ruff a directory of files).

Reply:
I trued to make it short and concise, with “designed by ruff” I was trying to highlight, that it’s needed because of an intentional design decision from ruff - to not apply the excludes from the config when a list of files is passed instead of a directory. 
now that I’m thinking about it feels like that’s the right thing to do for a tool like ruff, like why would you explicitly pass it a list of file names and still want to exclude some of those… 
Anyways I’ll try to come up with a better comment tomorrow.

Reply:
Makes sense, I just don't want to send people off to the ruff docs. We might as well explain here the exact design issue in play.

Reply:
Works for me.

Reply:
Just note the typo: s/explicitely/explicitly/ :)

Reply:
thx for the hint, changes pushed